### ISSUE TITLE
cannot find symbol "import com.spotify.sdk.android.authentication.RNSpotifyPackage"

### ISSUE THREAD

**Issue by DanCroweNH** [lufinkey/react-native-spotify#102]
Hi,

I'm currently trying to build a fresh React Native project and install this library via yarn but I'm getting some issues when running `react-native run-android`.

I've built the React Native project (via `react-native init Test`) to a Pixel 2 AVD running Android API 28 and that runs ok. Then I installed this project with `yarn add rn-spotify-sdk` and I ran the 2 `react-native link` commands and amended the `build.gradle` files as per the docs but I'm getting the below error when I build to Android.

I haven't tested this with iOS yet although it looks like an Android specific problem so I guess that iOS will most likely run fine. I'm not familiar with native Android code/project structure so any help is massively appreciated!

- **React**: 16.6.3
- **React Native**: 0.58.4
- **RN Spotify SDK**: 1.1.13
- **Android API**: 28 (v9.0 Pie)
- **Gradle**: 4.10.1
- **Android Plugin Version**: 3.3.1

> Task :rn-spotify-sdk:compileDebugJavaWithJavac
Note: C:\Users\Dan\Desktop\Test\node_modules\rn-spotify-sdk\android\src\main\java\com\lufinkey\react\spotify\RNSpotifyModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :app:compileDebugJavaWithJavac FAILED
C:\Users\Dan\Desktop\Test\android\app\src\main\java\com\test\MainApplication.java:6: error: cannot find symbol
import com.spotify.sdk.android.authentication.RNSpotifyPackage;
                                             ^
  symbol:   class RNSpotifyPackage
  location: package com.spotify.sdk.android.authentication
C:\Users\Dan\Desktop\Test\android\app\src\main\java\com\test\MainApplication.java:28: error: cannot find symbol
            new RNSpotifyPackage(),
                ^
  symbol: class RNSpotifyPackage
2 errors

> FAILURE: Build failed with an exception.

**Comment by DanCroweNH** [comment]
I've just noticed issue #88 opened about a month ago for this which gives a fix for this problem.

It looks like `react-native link` generates the wrong import for `RNSpotifyPackage` in `MainApplication.java`. It should be `import com.lufinkey.react.spotify.RNSpotifyPackage`.

I've just changed this over and it compiles fine now.
I'll leave this issue open for now as I'm not sure if this is an issue with this project or with React Native's link command but the above is a solution until the link command works as expected, in any case!

**Comment by lufinkey** [comment]
This seems to be an issue with `react-native link` because this bug didn't start occurring until the latest react-native update. It's incredibly bizarre that it's using the auth library bundle rather than the actual project bundle..?

I'll leave this open though and try to investigate when I get the chance.

**Comment by chrisflora** [comment]
I have a similar problem @lufinkey but it doesn't appear to have the same fix.  I am attempting to upgrade my react-native install to 0.60.4 and get the following error:

```
> Task :app:processDebugGoogleServices
Parsing json file: /Users/myuser/Projects/myproj/android/app/google-services.json

> Task :app:compileDebugJavaWithJavac FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/4.10.1/userguide/command_line_interface.html#sec:command_line_warnings
225 actionable tasks: 11 executed, 214 up-to-date
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/Users/myuser/Projects/myproj/android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:51: error: cannot find symbol
import com.spotify.sdk.android.authentication.RNSpotifyPackage;
                                             ^
  symbol:   class RNSpotifyPackage
  location: package com.spotify.sdk.android.authentication
/Users/myuser/Projects/myproj/android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:102: error: cannot find symbol
      new RNSpotifyPackage()
          ^
  symbol:   class RNSpotifyPackage
  location: class PackageList
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
```

Any ideas what I should do to fix this?

I have verified that my MainApplication.java file has this:
`import com.lufinkey.react.spotify.RNSpotifyPackage;`

and does NOT have
`import com.spotify.sdk.android.authentication.RNSpotifyPackage;`

I am fairly new to react-native and android/ios development as a whole, so any assistance would be greatly appreciated. 

My package.json file has this:
"rn-spotify-sdk": "^1.2.9"

I have uninstalled and reinstalled many many times now and have not had any luck getting away from this error.

When I go into the android directory and run a `./gradlew clean` it does not appear to do anything.   If I do a gradle sync in Android Studio, it appears to work correctly:

```
completed successfully	13 s 28 ms
Run build	12 s 910 ms
Load build	5 s 980 ms
Evaluate settings	5 s 978 ms
Configure build	4 s 939 ms
Calculate task graph	1 s 205 ms
Run tasks	784 ms
```

Any helpful hints?

**Comment by ririvas** [comment]
@chrisflora Hi, I had a similar issue when upgrading to React Native 0.60. My only way around was to start from a fresh install. I created a new project and added the relevant packages. I also followed the install instructions outlined in issue [139](https://github.com/lufinkey/react-native-spotify/issues/139).

**Comment by grabbou** [comment]
This can be closed in favour of #139. Going to look into this.
